src/Microsoft.ServiceFabric.Actors/Runtime/ActorStateProviderHelper.cs

```diff
@@ -435,18 +432,24 @@ internal static IActorStateProvider CreateDefaultStateProvider(ActorTypeInformat
                 return Task.FromResult(actorQueryResult);
             }
 
-            // Skip the previous returned entries
-            while (currentActorCount < previousActorCount)
+            // Find continuation point for enumeration
+            if (continuationToken != null)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                enumHasMoreEntries = enumerator.MoveNext();
-                currentActorCount++;
+                long previousActorCount = 0L;
+                if (long.TryParse((string)continuationToken.Marker, out previousActorCount))

```

Comment:
If the ActorId is of ActorIdKind.Long type, then this condition would succeed which is incorrect.

Reply:
Sending actor presence key itself in the continuation token might help ?

Reply:
Modified the logic to send presenceStorageKey instead of actor ID